internal/envoy/bootstrap.go

```diff
@@ -27,11 +30,25 @@ import (
 	envoy_api_v2_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
 	bootstrap "github.com/envoyproxy/go-control-plane/envoy/config/bootstrap/v2"
 	matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/ptypes/any"
 	"github.com/projectcontour/contour/internal/protobuf"
 )
 
+// bootstrapConfigFile stores the path to Envoy's bootstrap configuration.
+const bootstrapConfigFile = "envoy.json"
+
+// sdsTLSCertificateFile stores the path to the SDS resource with Envoy's
+// client certificate and key for XDS gRPC connection.
+const sdsTLSCertificateFile = "xds-tls-certicate.json"
+
+// sdsValidationContextFile stores the path to the SDS resource with
+// CA certificates for Envoy to use for the XDS gRPC connection.
+const sdsValidationContextFile = "xds-validation-context.json"
+
 // Bootstrap creates a new v2 Bootstrap configuration.
-func Bootstrap(c *BootstrapConfig) *bootstrap.Bootstrap {
+func Bootstrap(c *BootstrapConfig) {
```

Comment:
I think that this function ought to be broken down into internal helpers so that we can keep the existing tests working, and add coverage for the new stuff.

Here's what I would recommend:
- break out a helper that generates the Bootstrap struct
- now you can make a map of file path to content; this can be used to update the tests
- rename this function to `WriteBootstrap` and update doc comment accordingly
- all you need to do here is loop through the map and write each pair
- return an error from here if any of the output writing fails

Reply:
Done!